### DISCUSSION TITLE
[Community Call] Agenda – June 21, 2022

### DISCUSSION THREAD

**Discussion by natemoo-re** [withastro/roadmap#211] [Announcements]
## To Discuss

- **Announcement** Welcome new maintainers! 🎉 @readonlychild @thepassle @Yan-Thomas @hippotastic
- **Announcement** Improvements from performance audit (@matthewp)
  - `astro-island` approach [#3571](https://github.com/withastro/astro/pull/3571) | [Twitter Thread](https://twitter.com/matthewcp/status/1538978579102457860)
  - Support re-exporting astro components containing client components [#3625](https://github.com/withastro/astro/pull/3625)
    - __Before__ 928 page(s) built in 170.82s
    - __After__ 928 page(s) built in 114.65s
    - __32%__ faster
- **Announcement** Recursive Partial Hydration [Twitter Thread](https://twitter.com/n_moore/status/1539086497030316034) (@natemoo-re)
- **Action Item** Please give feedback on [#210](https://github.com/withastro/rfcs/pull/210) so we can discuss soon
- **Discussion** Expose named slots to renderers [#3652](https://github.com/withastro/astro/pull/3652)
- **Discussion** Rename `getHeaders()` to `getHeadings()` [#3627](https://github.com/withastro/astro/pull/3627)
- **Issues** [`p2-to-be-discussed`](https://github.com/withastro/astro/issues?q=is%3Aopen+is%3Aissue+label%3Ap2-to-be-discussed)

_If there is something that you would like to see discussed, please comment below. We will prioritize items already on the agenda, but if there is time to spare we will open up the call to get attention on and discuss [other threads and feature requests](https://github.com/withastro/rfcs/discussions)._

**Comment by FredKSchott** [comment]
> Support re-exporting astro components containing client components https://github.com/withastro/astro/pull/3625

Don't forget this quote from @hippotastic on Discord: 

> Oh wow, build performance has gone through the roof with this update!
> I can now build full Astro Docs in 22s rather than 184s before the upgrade!
> Hope that there's no hidden catch somewhere, it almost sounds too good to be true 😆

**Comment by delucis** [comment]
# Community Call Minutes, 21 June 2022

### News 📰

@natemoo-re reports what’s new in the Astroverse:

- Exciting news! Welcome 4 new Astro maintainers: @Yan-Thomas, @hippotastic, @readonlychild & @thepassle 🥳

- Big performance improvements this week thanks to @matthewp 🙌
  - includes moving from `<astro-root>` to `<astro-island>` 🏝️ and not duplicating hydration scripts
  - Builds **32% faster** (or more? one report of an 88% improvement 🤯)

- Draft PR withastro/astro#3652 to support named slots in renderers
  - Mostly internal changes, but impacts renderer API, so renderer authors may want to reach out about these changes

### RFCs/Issues to be discussed 💬

- Looked at RFC #791 / PR https://github.com/withastro/astro/pull/3627
    No objections ✅ 

- Discussion of withastro/astro#3618
  - Confusion around how these APIs behave in different environments and Static vs SSR
  - Document how to get `canonicalURL` without the actual feature

- Plan to update documentation around `injectScript` limitations: https://github.com/withastro/docs/issues/806

- withastro/astro#3573 — how to handle UTF-8 issues for SSG dev server. Follow up with @matthewp for thoughts. Dev server could serve UTF-8 by default, but could cause problems when built site hits a host that does _not_ handle that for you.

- Decided withastro/astro#3636 is a bug! 🐛 

- HMR not working for `<script>` tags — withastro/astro#3556
  - @natemoo-re: problem, no easy solve.
  - @ekafyi: Opt in to full page reload instead of HMR?
  - @tony-sull: Full page reload automatically if Astro detects a script like this. Already precedent for that in a few other cases.
  - @FredKSchott Agrees with Tony
  - @bholmesdev: Slinkity only HMRs for style & component assets, template changes reload everything
